sys/net/application_layer/coap/gcoap.c

```diff
@@ -135,7 +135,7 @@ static void _listen(sock_udp_t *sock)
         return;
     }
 
-    if (coap_get_code(&pdu) == COAP_CODE_EMPTY) {
+    if (pdu.hdr->code == COAP_CODE_EMPTY) {
```

Comment:
I agree that it would be valuable to improve this code. I avoid direct access to the struct when there is a nanocoap function (obviously ;-)) Does it make sense to create a `coap_is_empty()` inline function in nanocoap?